src/app/twelveLabs/actions.ts

```diff
@@ -34,6 +35,41 @@ export async function generateTranscriptions(videoId: string) {
   return transcriptionsSchema.parse(transcriptions);
 }
 
+// mb better promise.all then sort?
+export async function retrieveThumbnails(
+  parsedCaptions: Entry[],
+  videoId: string,
+) {
+  const schema = z.object({
+    thumbnail: z.string(),
+  });
+
+  const url = `https://api.twelvelabs.io/v1.2/indexes/${process.env.TWELVE_LABS_GLOBAL_INDEX_ID as string}/videos/${videoId}/thumbnail`;
+
+  const options = {
+    method: "GET",
+    headers: {
+      accept: "application/json",
+      "x-api-key": process.env.TWELVE_LABS_API_KEY as string,
+      "Content-Type": "application/json",
+    },
+  };
+
+  let thumbnails: string[] = [];
+
+  for (let i = 0; i < parsedCaptions.length; i++) {
+    const time = Math.floor(parsedCaptions[i].from / 1000);
+
+    const response = await fetch(`${url}?time=${time}`, options);
+
+    const { thumbnail } = schema.parse(await response.json());
+
+    thumbnails = [...thumbnails, thumbnail];
+  }
+
+  return thumbnails;
+}
```

Comment:
This code can run too long if a video is relatively long and cause serverless function timeout